main.go

```diff
@@ -739,6 +738,314 @@ func (u *UserSessionsApp) DeleteRequest(writer http.ResponseWriter, r *http.Requ
+		var search string
+		if err = rows.Scan(&search); err != nil {
+			return nil, err
+		}
+		retval = append(retval, search)
+	}
+
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+
+	return retval, nil
+}
+
+// insertSavedSearches adds new saved searches to the database for the user.
+func (se *SearchesDB) insertSavedSearches(username, searches string) error {
+	var (
+		err    error
+		userID string
+	)
+
+	query := `INSERT INTO user_saved_searches (user_id, saved_searches) VALUES ($1, $2)`
+
+	if userID, err = queries.UserID(se.db, username); err != nil {
+		return err
+	}
+
+	_, err = se.db.Exec(query, userID, searches)
+	return err
+}
+
+// updateSavedSearches updates the saved searches in the database for the user.
+func (se *SearchesDB) updateSavedSearches(username, searches string) error {
+	var (
+		err    error
+		userID string
+	)
+
+	query := `UPDATE ONLY user_saved_searches SET saved_searches = $2 WHERE user_id = $1`
+
+	if userID, err = queries.UserID(se.db, username); err != nil {
+		return err
+	}
+
+	_, err = se.db.Exec(query, userID, searches)
+	return err
+}
+
+// deleteSavedSearches removes the user's saved sessions from the database.
+func (se *SearchesDB) deleteSavedSearches(username string) error {
+	var (
+		err    error
+		userID string
+	)
+
+	query := `DELETE FROM ONLY user_saved_searches WHERE user_id = $1`
+
+	if userID, err = queries.UserID(se.db, username); err != nil {
+		return nil
+	}
+
+	_, err = se.db.Exec(query, userID)
+	return err
+}
+
+// SavedSearchesApp is an implementation of the App interface created to manage
+// saved-searches
+type SavedSearchesApp struct {
+	searches seDB
+	router   *mux.Router
+}
+
+// NewSearchesApp returns a new *SavedSearchesApp
+func NewSearchesApp(db seDB, router *mux.Router) *SavedSearchesApp {
+	searchesApp := &SavedSearchesApp{
+		searches: db,
+		router:   router,
+	}
+	router.HandleFunc("/searches/", searchesApp.Greeting).Methods("GET")
+	router.HandleFunc("/searches/{username}", searchesApp.GetRequest).Methods("GET")
+	router.HandleFunc("/searches/{username}", searchesApp.PutRequest).Methods("PUT")
+	router.HandleFunc("/searches/{username}", searchesApp.PostRequest).Methods("POST")
+	router.HandleFunc("/searches/{username}", searchesApp.DeleteRequest).Methods("DELETE")
+	router.Handle("/debug/vars", http.DefaultServeMux)
+	return searchesApp
+}
+
+// Greeting prints out a greeting to the writer from saved-searches.
+func (s *SavedSearchesApp) Greeting(writer http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(writer, "Hello from saved-searches.\n")
+}
+
+// GetRequest handles writing out a user's saved searches as a response.
+func (s *SavedSearchesApp) GetRequest(writer http.ResponseWriter, r *http.Request) {
+	var (
+		username   string
+		userExists bool
+		err        error
+		ok         bool
+		searches   []string
+		v          = mux.Vars(r)
+	)
+
+	if username, ok = v["username"]; !ok {
+		badRequest(writer, "Missing username in URL")
```

Comment:
This is my fault, but I'll mention it here so that it can get fixed in the future. It turns out that Go already has a function that does what badRequest() does. 

https://golang.org/pkg/net/http/#Error

Again, doesn't need to be fixed now, but we should probably file an issue in Jira with a priority of Minor so that it gets switched over to the http.Error() function at some point.

Reply:
Also in Jira now as CORE-9197. :)